src/Illuminate/Foundation/Console/VendorPublishCommand.php

```diff
@@ -158,13 +159,16 @@ protected function parseChoice($choice)
     protected function publishTag($tag)
     {
         $published = false;
+        $pathsToPublish = $this->pathsToPublish($tag);
 
-        foreach ($this->pathsToPublish($tag) as $from => $to) {
+        foreach ($pathsToPublish as $from => $to) {
             $this->publishItem($from, $to);
 
             $published = true;
         }
 
+        $this->laravel['events']->dispatch(new VendorTagPublished($tag, $pathsToPublish));
```

Comment:
There's a couple of concerns here: first of all it's unsure if anything is published at this point. As such it's better that this line is part of the `else` condition of the `if` statement below.

Secondly, the `$pathsToPublish` isn't necessarily all of the paths that were published (see the contents of `publishItem`).

Reply:
@ryangjchandler I answered this before reading your main PR comment, sorry.

Reply:
@driesvints No worries, wanted to get your opinion on it because the console output is actually a bit misleading.

An alternative solution might be only passing the `$tag` through to the event, technically speaking if you're listening for a particular tag, then you are 99% of the time going to know what _might_ have been published.

What do you think?

Reply:
@driesvints Okay, I've moved the event into the `else` arm and also updated the comment on the `$paths` property, so instead of stating that the array contains the paths that **were** published, it instead states that the `$paths` property holds all of the publishable paths registered by that tag.

Is this better?

Reply:
@ryangjchandler I'll let Taylor review this one, thanks.